cli/011-cli-multi-config.md

```diff
@@ -0,0 +1,74 @@
+- Start Date: 2021-08-22
+- RFC PR:
+- Issue:
+
+### Configuring multiple environments
+
+##### Problem
+
+Users may have multiple Temporal service environments (development, production, ..) with their own respective mTLS configs, namespaces, address etc. Current implementation of tctl config feature only supports configuring a single environment and doesn't help at quickly switching between multiple.
+
+##### Proposal
+
+To make it easy for users to switch between environments, allow configuring multiple instead of just one and provide command to switch between them.
+
+Syntax to activate environment configs:
+
+```shellsession
+$ tctl config use-environment <name>
+Active environment: <name>
+```
+
+If the environment doesn't yet exist in config, create it and set as active
+
+```shellsession
+$ tctl config use-environment <name>
+Environment <name> record doesn't exist, adding..
+Active environment: <name>
+```
+
+Split configuration keys into 1) environment scope - keys residing in a scope of specific environments 2) global scope - out of any specific environment
+
+Global scope keys:
+
+- "active" - name of active environment
+- "environments" - sequence of environments
+- "alias" - sequence of command aliases
+
+Environment scope keys:
+
+- namespace
+- address
+- data-converter-plugin
```

Comment:
Nit: IMO it's nice if these keys map directly to `tctl` flag names, eg. `data_converter_plugin`. This allows users to look at `tctl -h` as a quick reference for which config options are available.

Reply:
Ah, nevermind I see there's a proposal for changing the flag naming conventions. https://github.com/temporalio/proposals/blob/master/cli/005-cli-hyphens.md